core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -1241,6 +1243,81 @@ public static Expression multiplyDivide(Expression e, BigDecimal multiplier,
     }
   }
 
+  /** Implementor for the {@code MODE} aggregate function. */
+  static class ModeImplementor extends StrictAggImplementor {
+    @Override protected void implementNotNullReset(AggContext info,
+        AggResetContext reset) {
+
+      reset.currentBlock().add(
+          Expressions.statement(
+              Expressions.assign(reset.accumulator().get(0),
+                  Expressions.constant(null))));
+
+      reset.currentBlock().add(
+          Expressions.statement(
+              Expressions.assign(reset.accumulator().get(1),
+                  Expressions.new_(HashMap.class))));
+
+      reset.currentBlock().add(
+          Expressions.statement(
+              Expressions.assign(reset.accumulator().get(2),
+                  Expressions.constant(0, Long.class))));
+    }
+
+    @Override protected void implementNotNullAdd(AggContext info,
+        AggAddContext add) {
+      Expression args = add.arguments().get(0);
+
+      Expression accResult = add.accumulator().get(0);
+      Expression accMap = add.accumulator().get(1);
+      Expression accLong = add.accumulator().get(2);
```

Comment:
Maybe better variable names could be used here, and some comments could be added to improve readability.

Reply:
Thank you very much for your advice. I will make some improvements